tests/test_odl.py

```diff
@@ -1,58 +1,56 @@
-import pytest
-import os
-import json
 import datetime
-import dateutil
-import re
+import json
+import os
 import urllib.parse
-from pdb import set_trace
-from core.testing import DatabaseTest
-from core.metadata_layer import TimestampData
+
+import dateutil
+import pytest
+from dateutil.tz import tzoffset
+from freezegun import freeze_time
+from mock import MagicMock
+
+from api.circulation_exceptions import *
+from api.odl import (
+    ODLAPI,
+    MockODLAPI,
+    MockSharedODLAPI,
+    ODLExpiredItemsReaper,
+    ODLHoldReaper,
+    ODLImporter,
+    SharedODLAPI,
+    SharedODLImporter,
+)
 from core.model import (
     Collection,
     ConfigurationSetting,
-    Credential,
     DataSource,
     DeliveryMechanism,
     Edition,
     ExternalIntegration,
     Hold,
     Hyperlink,
-    Identifier,
     Loan,
     MediaTypes,
     Representation,
     RightsStatus,
-    get_one,
-)
-from api.odl import (
-    ODLImporter,
-    ODLHoldReaper,
-    MockODLAPI,
-    SharedODLAPI,
-    MockSharedODLAPI,
-    SharedODLImporter,
-)
-from api.circulation_exceptions import *
-from core.util.datetime_helpers import (
-    datetime_utc,
-    strptime_utc,
-    utc_now,
-)
-from core.util.http import (
-    BadResponseException,
-    RemoteIntegrationException,
 )
+from core.scripts import RunCollectionMonitorScript
+from core.testing import DatabaseTest
+from core.util import datetime_helpers
+from core.util.datetime_helpers import datetime_utc, utc_now
+from core.util.http import BadResponseException, RemoteIntegrationException
 from core.util.string_helpers import base64
 
+
 class BaseODLTest(object):
     base_path = os.path.split(__file__)[0]
     resource_path = os.path.join(base_path, "files", "odl")
 
     @classmethod
     def get_data(cls, filename):
         path = os.path.join(cls.resource_path, filename)
-        return open(path, "rb").read()
+        return open(path, "r").read()
+
```

Comment:
After looking at the core changes (specifically https://github.com/ThePalaceProject/circulation-core/pull/16) I see that this works because the eventual consumer of this data has been changed to handle either bytes or strings. That's a good change but since real data from HTTP responses will come in as bytes I think the test was more realistic before.